### ISSUE TITLE
Support regex highlighting on documentation/signature help

### ISSUE THREAD

**Issue by SourishBiswas23** [Saghen/blink.cmp#336]
### Feature Description

Currently, blink.nvim updates on every keystroke .While this provides a smooth experience, it leads to high CPU usage on my system. The CPU usage reaches 25-30% during typing.

In nvim-cmp, i added a debounce time of 300ms for my typing speed of 65 wmp and observed that the cpu usage went down to 10-12% while the completion was fast enough.

Proposed Feature:

Add a configurable debounce time for updates.
    Default: 0ms (retain current behavior for users who prioritize immediate updates).
    Configurable Option: Allow users to set a custom debounce time, such as .

Benefits:

    Significantly lower CPU usage, particularly on resource-constrained systems.
    Provides flexibility for users to balance responsiveness with system performance.

**Comment by Saghen** [comment]
Makes sense, what are your system specs? Do you happen to be using `auto_show = true` on documentation, in which case, does disabling that reduce your CPU usage? I'd like for blink to be fast enough as-is for low powered systems but a debounce option makes sense to me regardless

**Comment by SourishBiswas23** [comment]
The CPU in my laptop is an Intel(R) Core(TM) i5-1035G1 (8) @ 3.60 GHz with 12 GB of RAM.

I need the documentation to always be available because there are cases where prefixes expand into multiple snippets. For example, typing vfor expands into a Vue directive (the required one) and also a vfor HTML tag (emitted by Emmet). The documentation is the only way to differentiate between such completions.

I tried the other options, and while they reduce CPU usage slightly, the decrease in load is not very significant.

**Comment by Saghen** [comment]
Thanks, sounds like it's the treesitter highlighting on the documentation that's slowing things down. I've added an option to disable it on the documentation and signature help with:

```
windows.documentation.treesitter_highlighting = false
windows.signature_help.treesitter_highlighting = false
```

Hopefully that speeds things up on your end, although I'd like to keep this issue open to track regex highlighting

**Comment by Saghen** [comment]
For the sake of simplicity, I'm not planning to implement regex based highlighting. On resource constraints systems, you may disable `treesitter_highlighting` instead
